python-checks/src/main/java/org/sonar/python/checks/SecureModeEncryptionAlgorithmsCheck.java

```diff
@@ -0,0 +1,135 @@
+    PYCRYPTO_VULNERABLE_MODES_BY_API.put("Crypto.Cipher.Blowfish.new", Arrays.asList("Crypto.Cipher.Blowfish.MODE_ECB", "Crypto.Cipher.Blowfish.MODE_CBC"));
+    PYCRYPTO_VULNERABLE_MODES_BY_API.put("Crypto.Cipher.AES.new", Arrays.asList("Crypto.Cipher.AES.MODE_ECB", "Crypto.Cipher.AES.MODE_CBC"));
+  }
+
+  private static final Map<String, List<String>> CRYPTODOMEX_VULNERABLE_MODES_BY_API = new HashMap<>();
+  static {
+    CRYPTODOMEX_VULNERABLE_MODES_BY_API.put("Cryptodome.Cipher.DES.new", Arrays.asList("Cryptodome.Cipher.DES.MODE_ECB", "Cryptodome.Cipher.DES.MODE_CBC"));
+    CRYPTODOMEX_VULNERABLE_MODES_BY_API.put("Cryptodome.Cipher.CAST.new", Arrays.asList("Cryptodome.Cipher.CAST.MODE_ECB", "Cryptodome.Cipher.CAST.MODE_CBC"));
+    CRYPTODOMEX_VULNERABLE_MODES_BY_API.put("Cryptodome.Cipher.DES3.new", Arrays.asList("Cryptodome.Cipher.DES3.MODE_ECB", "Cryptodome.Cipher.DES3.MODE_CBC"));
+    CRYPTODOMEX_VULNERABLE_MODES_BY_API.put("Cryptodome.Cipher.ARC2.new", Arrays.asList("Cryptodome.Cipher.ARC2.MODE_ECB", "Cryptodome.Cipher.ARC2.MODE_CBC"));
+    CRYPTODOMEX_VULNERABLE_MODES_BY_API.put("Cryptodome.Cipher.Blowfish.new", Arrays.asList("Cryptodome.Cipher.Blowfish.MODE_ECB", "Cryptodome.Cipher.Blowfish.MODE_CBC"));
+    CRYPTODOMEX_VULNERABLE_MODES_BY_API.put("Cryptodome.Cipher.AES.new", Arrays.asList("Cryptodome.Cipher.AES.MODE_ECB", "Cryptodome.Cipher.AES.MODE_CBC"));
+  }
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(CALL_EXPR, ctx -> {
+      CallExpression callExpression = (CallExpression) ctx.syntaxNode();
+      Symbol calleeSymbol = callExpression.calleeSymbol();
+      if (calleeSymbol == null) {
+        return;
+      }
+      checkPycaLibrary(ctx, callExpression, calleeSymbol);
+      checkPycryptoAndCryptodomeLibraries(ctx, callExpression, calleeSymbol);
+      checkPydesLibrary(ctx, callExpression, calleeSymbol);
+    });
+  }
+
+  private static void checkPydesLibrary(SubscriptionContext ctx, CallExpression callExpression, Symbol calleeSymbol) {
+    if ("pyDes.des".equals(calleeSymbol.fullyQualifiedName())) {
+      ctx.addIssue(callExpression.callee(), MESSAGE);
+    }
+  }
+
+  private static void checkPycryptoAndCryptodomeLibraries(SubscriptionContext ctx, CallExpression callExpression, Symbol calleeSymbol) {
+    if ("Crypto.Cipher.PKCS1_v1_5.new".equals(calleeSymbol.fullyQualifiedName()) || "Cryptodome.Cipher.PKCS1_v1_5.new".equals(calleeSymbol.fullyQualifiedName())) {
+      ctx.addIssue(callExpression.callee(), MESSAGE);
+      return;
+    }
+
+    List<String> vulnerableModes = PYCRYPTO_VULNERABLE_MODES_BY_API.getOrDefault(calleeSymbol.fullyQualifiedName(),
+      CRYPTODOMEX_VULNERABLE_MODES_BY_API.get(calleeSymbol.fullyQualifiedName()));
+
+    if (vulnerableModes != null) {
+      List<Argument> arguments = callExpression.arguments();
+      if (arguments.size() < 2) {
+        return;
+      }
+      Argument argument = arguments.get(1);
```

Comment:
From my understanding, the API from PyCryptodome (when imported either as `Crypto` or `Cryptodome`) can also be called with a keyword argument for `mode` (see https://pycryptodome.readthedocs.io/en/latest/src/cipher/des.html for example).
Maybe a call such as `new(mode = ..., key = ...)` would be quite marginal and we don't need to bother to support it, but I just wanted to point it out.